### PR TITLE
Decisions: ActionForm, Submissions & Sagas for Simple Decisions

### LINKED ISSUE #3750: Decisions: Connect Preview page submission
### Story

As a user after providing Decision details, I want to be able to submit the Decision.

## Implementation

- Handle loading during transaction
- Create Action & saga (Decision is almost identical to a Motion, using same contract, but without an on chain Action)
- Save Decision title & details to IPFS
- Call saga, save IPFS hash
- Route to staking motions page

### PR DESCRIPTION
**Pull request by danbr**
## Description

This PR:

- Wraps ColonyDecisionPreview in an ActionForm.
- Removes `ColonyActions.Decision`, instead uses `ColonyMotions.CreateDecisionMotion`.
- Implements the Saga for Decisions 
- Stores Decision details in IPFS
- The Decision IPFS hash is stored instead of the annotation hash, by a standard motion.

Not included:

- Update transactions dialog to use more appropriate method names.

For some background:

- From the Contract (VotingReputation.sol) perspective, a Decision is identified from a standard motion by passing `0x12345678` at creation. 
- A Decision is a motion that does not have an on-chain action.
- A Decision is not `(MotionState.)Finalizable`

Please note, this PR (and soon, the feature branch) will be using an updated version of ColonyNetwork (and ColonyClient) therefore it may be necessary to run:
```
npm run provision
```
to use the updated contract that supports Decisions, and also `provision`-ing will be needed when switching back to non-Decisions Dapp work.

*Please have the Governance extension installed, the Reputation miner running, and have some reputation.

Resolves #3750

### CODE REVIEW

**Review by willm30 — COMMENTED** [review]
@danbr I am getting the following error. I have done `npm install`, `npm run provision` and I've restarted my dev environment, but still receiving it. Is there anything else I can check to verify if my env is set up correctly? 

inval-arr

**Comment by willm30** [comment]
> @danbr I am getting the following error. I have done `npm install`, `npm run provision` and I've restarted my dev environment, but still receiving it. Is there anything else I can check to verify if my env is set up correctly?
> 

@danbr I didn't have the reputation miner turned on. I turned that on and it fixed it. 

I am being redirected to the pink naked mole, though. I guess that's being handled in a separate pr?

**Comment by danbr** [comment]
> > @danbr I am getting the following error. I have done `npm install`, `npm run provision` and I've restarted my dev environment, but still receiving it. Is there anything else I can check to verify if my env is set up correctly?
> 
> @danbr I didn't have the reputation miner turned on. I turned that on and it fixed it.
Great - its also a good point. I will include this in the PR testing description.....

> I am being redirected to the pink naked mole, though. I guess that's being handled in a separate pr?
Yea exactly. This PR grew too big, so i drew the line at 'the pink naked mole'. Seemed like a good place!

**Review by willm30 — APPROVED** [review]
Looking good / working well for me. 

Just noting the following warning in the console when you click "Publish": 
formwarn

**Comment by danbr** [comment]
> Looking good / working well for me.
> 
> Just noting the following warning in the console when you click "Publish": formwarn

Thank you.... Its not connected with Decisions work.

**Review by chinins — COMMENTED** [review]
It's all good. I would just like to have a look after you rebase it on the latest simple-decisions branch - to make sure it's all cleaned up.

Just let me know once rebased and I'll review it very quickly.

File: src/modules/dashboard/sagas/motions/createDecisionMotion.ts [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+
+    const details: DecisionDetails = {
+      title: decisionTitle,
+      description: decisionDescription,
+      domainId,
+    };
+    /*
+     * Upload Decision details to IPFS
+     */
+    const decisionIpfsHash = yield call(ipfsUpload, JSON.stringify(details));
```
[the remark this answers is not in the document]
Comment (danbr) [reply src/modules/dashboard/sagas/motions/createDecisionMotion.ts:148]: I just rebased the feature branch with Master, and then this PR with the feature branch, only to discover that IPFS changes have not been merged into Master yet.

File: src/modules/dashboard/components/DecisionPreview/DecisionPreviewForm.tsx [hunk 2]
```diff
@@ -35,26 +38,29 @@ const MSG = defineMessages({
...
+    notified of things happening in our Colony.</p>`,
+  domainId: 1,
 };
 
-const handleSubmit = () => {};
-
 const displayName = 'dashboard.DecisionPreview';
 
-const DecisionPreview = () => {
+const DecisionPreviewForm = () => {
```
[the remark this answers is not in the document]
Comment (danbr) [reply src/modules/dashboard/components/DecisionPreview/DecisionPreviewForm.tsx:57]: I dont have a strong opinion as the `form` is only wrapping a button, and not the whole page as originally expected. So i'm happy to change.

File: src/modules/dashboard/components/DecisionPreview/DecisionPreview.tsx [hunk 3]
```diff
@@ -50,33 +50,44 @@ const MSG = defineMessages({
...
+        decisionTitle: decisionData?.title,
+        decisionDescription: decisionData?.description,
+        motionDomainId: decisionData?.motionDomainId,
+      })),
+      withMeta({ history }),
+    ),
+    [colonyData],
+  );
+
+  // console.log(`🚀 ~ mapPayload ~ decisionData`, decisionData);
```
Comment (chinins) [inline src/modules/dashboard/components/DecisionPreview/DecisionPreview.tsx:85]: Should this whole block be deleted?

**Review by chinins — APPROVED** [review]
Very good, just please delete the commented out block that I think won't be used anywhere.

**Pull request merged**